src/library.js

```diff
@@ -1733,6 +1733,7 @@ mergeInto(LibraryManager.library, {
   $readSockaddr__deps: ['$Sockets', '$inetNtop4', '$inetNtop6', 'ntohs'],
   $readSockaddr: function (sa, salen) {
     // family / port offsets are common to both sockaddr_in and sockaddr_in6
+    {{{ from64('sa') }}}
```

Comment:
The preferred way to do these conversions is by specifying signatures for the JS functions that take addresses.  In this case I think it means that easy of the library entry points that calls `readSockAddr` can/should specify a `__sig` attribute with `p` in the positions that correspond to pointer-width/size_t-width things.

Reply:
Ah, I was not aware that `__sig` handles conversion, too. Thanks for pointing it out.